Yoast/javascript
# Prepare Arabic readability for 15.0 release

## Summary


This PR can be summarized in the following changelog entry:
* [yoastseo] Adds passive voice assessment for Arabic, including a check, a number or specs and a list of Arabic passive verbs with long vowel in their root.
* [yoastseo] Adds the consecutive sentences assessment for Arabic.
* [yoastseo] Implements the transition words assessment for Arabic.
* [yoastseo] Adds unit tests for the Arabic sentence length assessment.

* [Yoast Seo Free] Adds an assessment for the recognition of the passive voice in Arabic texts (for example: أُنشئت الشركة في الثمانينيات or ودعيت هذه المجموعات للإدلاء بملاحظات ختامية في الجلسة الأخيرة).
* [Yoast SEO Free] Adds the consecutive sentences assessment for Arabic.
* [Yoast SEO Free] Adds the assessment for transition words in Arabic.

## Relevant technical choices:

* There is not much readability information about Arabic, specifically not about sentence length. [Our research](https://github.com/Yoast/research/wiki/Sentence-Length-Arabic) showed that in terms of sentence length Arabic is not much different from English. We made a decision to keep the default value of 20 words per sentence as a guideline for the assessment. 

## Test instructions

This PR can be tested by following these steps:

* Set the WP environment language to Arabic
* Test these functionalities in WordPress SEO Free

* Test the passive voice assessment:
    * You should see a bullet assessment for passive verb forms in the `Readability` section
    * Add these three sentences to your post:
    `.فُرش البيت الجديد`
    `.حماسي في تلك الأيام أثير كلياً بواسطة الكحول`
    `.لربّما هو سوعد`
    * The passive voice bullet should turn red

* Test the consecutive sentences assessment:
   * Write three or more Arabic sentences that meet the following requirements:
     * all begin with the same written-out number, like `سبعة` ("seven")
        * E.g. `.سبعة صوتا مقابل خمسة ضد، وامتناع واحد. سبعة صمامات مختلفة يتم إنتاجها في المبنى. سبعة متزوجين, واثنان في علاقة مع بعضهما`
     * all begin with the same demonstrative pronoun, like `هذا` ("this").
        * E.g. `.هذا الكهف مسكون. هذا التطبيق مكسور. هذا التطبيق مكسور`
   * In each of the above cases, the sentence beginning readability assessment should read: `Consecutive sentences: There is enough variety in your sentences. That's great!`
   * Write three or more Arabic sentences that begin with the same word.
       * E.g. `.مرحبا ، اهلا وسهلا بكم. مرحبا كيف حالك؟ مرحبا هل انت في المنزل. مرحبا لجميع الوافدين الجدد`
   * The sentence beginning readability assessment should read: Consecutive sentences: `The text contains 4 consecutive sentences starting with the same word. Try to mix things up!`

* Test the transition words assessment:
    * Paste in this text containing the transition word `لاسيما`:
       `.السفر بالقطار مريح لاسيما على الدرجة الأولى`
    * The sentence should be pasted until the assessment bullet turns green (above 29% of the sentences) and read: `Transition words: Well done!`
    * All the sentences containing the transition word should be highlighted.
    
    * Paste in this text containing the transition word `وفي الوقت نفسه`:
       `.سأفعل البقالة. وفي الوقت نفسه ، يمكنك الدراسة`
    * The sentence should be pasted until the assessment bullet turns green (above 29% of the sentences) and read: `Transition words: Well done!`
    * Out of the two sentences, only the second should be highlighted.
    
    * Paste in this text containing the transition word `...كل من ... و`:
       `.يمكننا الحصول إما على الخبز أو الأرز`
    * The sentence should be pasted until the assessment bullet turns green (above 29% of the sentences) and read: `Transition words: Well done`!
    * The whole sentence should be highlighted.

* Test the sentence length assessment:
   * Note that this PR is not introducing any user-facing changes in the behaviour of the sentence length assessment in Arabic.
   * Enter one Arabic sentence shorter than 20 words. The sentence length assessment should return a good score.
      * E.g.: `.تختفي الأعراض بين النوبات بالكامل، لكن المشاكل العصبية الدائمة تحدث في أغلب الأحيان خصوصاً إذا كان المرض في مراحل متقدمة`
   * Enter one Arabic sentence longer than 20 words. The sentence length assessment should return a bad score.
      * E.g.: `.التصلب المتعدد، ويعرف بأسماء عديدة، منها التصلّب اللويحي والتصلب المنتثر أو التهاب الدماغ والنخاع المنتثر، وهو التهاب ينتج عن تلف الغشاء العازل للعصبونات في الدماغ والحبل الشوكي`

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes https://yoast.atlassian.net/browse/LIN-566
